alternator/streams.cc

```diff
@@ -593,6 +602,12 @@ future<executor::request_return_type> executor::describe_stream(client_state& cl
 
                 last = std::nullopt;
             }
+
+            if (limit > 0) {
+                // #7409 - shards must be returned in lexicographical order,
+                // so only sort for prevs sake - i.e. finding parent
+                std::stable_sort(sv.streams.begin(), sv.streams.end(), token_cmp);
```

Comment:
I don't understand what I'm seeing here... Didn't you already sort the same array by id_cmp? Why do we need to resort it using token_cmp?  The comment seems to contradict what you do (you say "must be returned in lexicographical order" and then sort differently)...
Also, what does the "limit > 0" test mean?

Reply:
We need to return shards in "byte representation" order (lexical). But when we process next generation and this is "prev", we want things in token order, so we can quickly find the parent of a shard (if any). 
The limit check is simply to bypass the sort if we are with this paged call - no need to sort if we are not processing anything else.

Reply:
Simplified.